Source/hwcursor.cpp

```diff
@@ -29,6 +29,27 @@ enum class HotpointPosition {
 	Center,
 };
 
+Size ScaledSize(Size size)
+{
+	if (renderer != nullptr) {
+		float scaleX;
+		float scaleY;
+		SDL_RenderGetScale(renderer, &scaleX, &scaleY);
+		size.width *= scaleX;  // NOLINT(bugprone-narrowing-conversions)
+		size.height *= scaleY; // NOLINT(bugprone-narrowing-conversions)
+	}
+	return size;
+}
+
+bool IsCursorSizeAllowed(Size size)
+{
+	if (sgOptions.Graphics.nHardwareCursorMaxSize <= 0)

```

Comment:
Might be more logic to have 0 mean that the cursor should always be software rendered.